### DISCUSSION TITLE
Zustand/Context

### DISCUSSION THREAD

**Discussion by httpete** [pmndrs/zustand#1331] [Q&A]
Hi,

I use the Zustand/Context right from you. I see that you store the createStore() in a useRef. I bet in most SPA uses cases, this is always desirable.  In my case, I want to be able to call root.render() again on an already mounted node (thus a logical update) and I find that Zustand/Context (by design ) doesn't rerender the Context with a newly updated store.

My question is why bother at all with a useRef<>() ? Doesn't React.Context take care of this already natively?

**Comment by dai-shi** [comment]
With zustand v4, you can use new apis instead of zustand/context. It's up to you whether to use useRef or not.

**Comment by httpete** [comment]
Thank you @dai-shi - We initialize values to the store from the server so the Context suits this perfectly. 
I understand how it's up to me, I just wanted to know the rationale behind the decision to useRef for the context value. If I understand properly, that essentially freezes the value of the context to the first function passed in. 
It is very possible that in a typical SPA use case, this is desirable.  Thank you this is just for my learning of how Zustand works and if there is any underlying interaction with React.Context that I should be aware of.

**Reply by dai-shi** [reply]
The rationale is you need a stable reference of the store. If you create a store reference in the render function and forget, it will cause infinite loop. To get stable reference, useRef, useMemo or `use-constant` should do.
